### PR TITLE
Fix #7174: ArgumentNullException for NSExpression.FromConstant

### LINKED ISSUE #7174: NSExpression.FromConstant - not allow to pass null
### Steps to Reproduce
1. Try to create a new expression `NSExpression.FromConstant(null)`

### Expected Behavior
A new NSExpression instance is created

### Actual Behavior
ArgumentNullException is thrown

### Environment
VS for Mac or Jetbrains Rider
```
=== Visual Studio Community 2019 for Mac ===

Version 8.3.1 (build 18)
Installation UUID: 92984a18-8886-429d-820e-76a61eb80fb6
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 5.16.1.24 (d16-3 / 08809f5b)

	Package version: 604000198

=== Mono Framework MDK ===

Runtime:
	Mono 6.4.0.198 (2019-06/fe64a4765e6) (64-bit)
	Package version: 604000198

=== NuGet ===

Version: 5.3.0.6192

=== .NET Core SDK ===

SDK: /usr/local/share/dotnet/sdk/3.0.100/Sdks
SDK Versions:
	3.0.100
	2.2.401
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/mono/msbuild/Current/bin/Sdks

=== .NET Core Runtime ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	3.0.0
	2.2.6
	2.1.13

=== Xamarin.Profiler ===

Version: 1.6.12.26
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Apple Developer Tools ===

Xcode 11.0 (14936)
Build 11A420a

=== Xamarin.Mac ===

Version: 6.2.0.42 (Visual Studio Community)
Hash: 5e8a208b
Branch: d16-3
Build date: 2019-09-17 14:19:30-0400

=== Xamarin.iOS ===

Version: 13.2.0.42 (Visual Studio Community)
Hash: 5e8a208b
Branch: d16-3
Build date: 2019-09-17 14:19:30-0400

=== Xamarin Designer ===

Version: 16.3.0.244
Hash: e15b2ee97
Branch: remotes/origin/d16-3
Build date: 2019-09-25 00:03:16 UTC

=== Xamarin.Android ===

Version: 10.0.0.43 (Visual Studio Community)
Commit: xamarin-android/d16-3/8af1ca8
Android SDK: /Users/tuyen/Library/Android/sdk
	Supported Android versions:
		8.0 (API level 26)
		8.1 (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 29.0.4
SDK Build Tools Version: 29.0.2

Build Information: 
Mono: mono/mono/2019-06@7af64d1ebe9
Java.Interop: xamarin/java.interop/d16-3@5836f58
LibZipSharp: grendello/LibZipSharp/d16-3@71f4a94
LibZip: nih-at/libzip/rel-1-5-1@b95cf3f
ProGuard: xamarin/proguard/master@905836d
SQLite: xamarin/sqlite/3.27.1@8212a2d
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-3@cb41333

=== Microsoft Mobile OpenJDK ===

Java SDK: /Users/tuyen/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_1.8.0.25
1.8.0-25
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android SDK Manager ===

Version: 1.4.0.65
Hash: c33b107
Branch: remotes/origin/d16-3
Build date: 2019-09-19 20:42:44 UTC

=== Android Device Manager ===

Version: 1.2.0.115
Hash: 724ea69
Branch: remotes/origin/d16-3
Build date: 2019-09-19 20:43:06 UTC

=== Xamarin Inspector ===

Version: 1.4.3
Hash: db27525
Branch: 1.4-release
Build date: Mon, 09 Jul 2018 21:20:18 GMT
Client compatibility: 1

=== Build Information ===

Release ID: 803010018
Git revision: 272ab73f00ab2b13ec97dc8b31d02980cf66d47f
Build date: 2019-09-27 19:26:51+00
Build branch: release-8.3
Xamarin extensions: 451b00e65052c11c16b79e6d31a24c4cf0d1b794

=== Operating System ===

Mac OS X 10.14.6
Darwin 18.7.0 Darwin Kernel Version 18.7.0
    Tue Aug 20 16:57:14 PDT 2019
    root:xnu-4903.271.2~2/RELEASE_X86_64 x86_64


```

### Build Logs
Not required

### Example Project (If Possible)
Not required

> VS bug [#993925](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/993925)

### PR DESCRIPTION
**Pull request by tuyen-vuduc**
Fix #7174 
- Add missing NullAllowed attribute
- Add unit test for null value

### CODE REVIEW

File: src/foundation.cs [hunk 1]
```diff
@@ -4197,7 +4197,7 @@ interface NSException : NSSecureCoding, NSCopying {
 	[DisableDefaultCtor]
 	interface NSExpression : NSSecureCoding, NSCopying {
 		[Static, Export ("expressionForConstantValue:")]
-		NSExpression FromConstant (NSObject obj);
+		NSExpression FromConstant ([NullAllowed]NSObject obj);
```
Comment (rolfbjarne) [inline src/foundation.cs]: Coding style: space between attribute and type name.

File: tests/monotouch-test/Foundation/NSExpressionTest.cs [hunk 2]
```diff
@@ -45,13 +45,22 @@ void TestProperties (NSExpression expression, List<string> availableProperties)
...
 		}
 
-
-		[Test]
-		public void FromConstant ()
-		{
-			using (var expression = NSExpression.FromConstant (new NSString ("Foo")))
-			using (var result = expression.EvaluateWith (null, null) as NSString)
-				Assert.AreEqual ("Foo", result.ToString ());
+		[TestCase("Foo", Result = "Foo")]
```
Comment (rolfbjarne) [inline tests/monotouch-test/Foundation/NSExpressionTest.cs]: Coding style: space after space before opening parenthesis.

File: tests/monotouch-test/Foundation/NSExpressionTest.cs [hunk 3]
```diff
@@ -45,13 +45,22 @@ void TestProperties (NSExpression expression, List<string> availableProperties)
...
+		[TestCase(null, Result = null)]
+		public object FromConstant (object input)
+		{
+            NSObject value = null;
```
Comment (rolfbjarne) [inline tests/monotouch-test/Foundation/NSExpressionTest.cs]: Coding style: indent using tabs instead of spaces.

File: tests/monotouch-test/Foundation/NSExpressionTest.cs [hunk 4]
```diff
@@ -45,13 +45,22 @@ void TestProperties (NSExpression expression, List<string> availableProperties)
...
+
+            switch (input)
+            {
```
Comment (rolfbjarne) [inline tests/monotouch-test/Foundation/NSExpressionTest.cs]: Coding style: opening brace goes on the same line as the switch keyword (with a space to separate the closing parenthesis and the opening brace).

File: tests/monotouch-test/Foundation/NSExpressionTest.cs [hunk 5]
```diff
@@ -45,13 +45,22 @@ void TestProperties (NSExpression expression, List<string> availableProperties)
...
+                case String stringValue:
+                    value = new NSString(stringValue);
```
Comment (rolfbjarne) [inline tests/monotouch-test/Foundation/NSExpressionTest.cs]: Coding style: space before opening parenthesis.

File: tests/monotouch-test/Foundation/NSExpressionTest.cs [hunk 6]
```diff
@@ -45,13 +45,22 @@ void TestProperties (NSExpression expression, List<string> availableProperties)
...
+                    break;
+            }
+
+            using (var expression = NSExpression.FromConstant(value))
+            using (var result = expression.EvaluateWith(null, null) as NSObject)
+                return result?.ToString();
```
Comment (rolfbjarne) [inline tests/monotouch-test/Foundation/NSExpressionTest.cs]: Coding style: space before opening parenthesis (in the previous two lines as well).

**Comment by rolfbjarne** [comment]
build

**Comment by rolfbjarne** [comment]
Thanks a lot for your contribution!
